manager/manager.go

```diff
@@ -783,6 +790,35 @@ func (m *manager) registerCollectors(collectorConfigs map[string]string, cont *c
 	return nil
 }
 
+// Enables overwriting an existing containerData/Handler object for a given containerName.
+// can't use createContainer as it just returns if a given containerName has a handler already
+// Ex: rkt handler will want to take priority over the raw handler, but the raw handler might be created first
+
+// Only allow raw handler to be overriden
+func (m *manager) overrideContainer(containerName string, watchSource watcher.ContainerWatchSource) error {
```

Comment:
+1. I don't think the "override" logic should be part of the watcher. The watcher sees a new container, and reports an "add" event, and then I think the manager should have the logic to say either, "This already exists, ignore it" or "This exists in raw form, replace it with the runtime-specific handler".